### PR TITLE
[1LP][RFR] Smart Management test

### PR DESCRIPTION
**Pull request by shalomnaim1**
{{pytest: cfme/tests/containers/test_smart_management.py -v --use-provider cm-env1 }}

Before the test starts the code remove all the previous smart management configuration

Add Smart management test for the following objects:
1) Template
2) Image Registry
3) Image
4) Node
5) Pod
6) Project
7) Containers Provider

The test choose random instance of object and do the following for each:
1) Navigate to smart management view
2) choose random category and tag
3) save configuration
4) verify configuration

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit a29a69d2687d9cd6a510f01f00ea4a3e2b068b51:

`cfme/tests/containers/test_smart_management.py`:
- :red_circle: Line 54:5: **N806** *variable in function should be lowercase*
- :red_circle: Line 59:5: **N806** *variable in function should be lowercase*
- :red_circle: Line 84:5: **N806** *variable in function should be lowercase*
- :red_circle: Line 85:5: **N806** *variable in function should be lowercase*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 36dff1ff808f494d01f0cfd609692dbaf3207c68:

`cfme/tests/containers/test_smart_management.py`:
- :grey_exclamation: Line 43:5: **T001** *print statement found.*
- :grey_exclamation: Line 49:5: **T001** *print statement found.*
- :red_circle: Line 58:5: **N803** *argument name should be lowercase*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by mfalesni** [comment]
@shalomnaim1 If you want to have it reviewed, you need to make the PR [RFR]

File: cfme/tests/containers/test_smart_management.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    obj_inst.summary.reload()
+    tag_display_text = obj_inst.summary.smart_management.my_company_tags.pop()
+    tag_display_text = tag_display_text.text_value
+
+    assert re.match(regex, tag_display_text), "Tag formatting is invalid! "
+    actual_tags_on_instance = obj_inst.get_tags()
+
+    # Validate tag seted successfully
+    assert len(actual_tags_on_instance) == 1, \
+        "Fail to set a tag for {type}".format(type=test_item.obj.__module__.title().split(".")[-1])
```
Comment (gshefer) [inline cfme/tests/containers/test_smart_management.py]: Use test_item.obj.name instead

**Comment by dajoRH** [comment]
Lint report for commit 9b1e3021b8eb0d734633d7528be08ec3c2810c00:

`cfme/tests/containers/test_smart_management.py`:
- :bangbang: Line 81:5: **E303** *too many blank lines (3)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by shalomnaim1** [comment]
@gshefer I need the name of the object not the value actual CFME instance name
as you can see below, 
>>> obj_inst.name
Out[1]: u'eap70-amq-s2i'
>>> type(obj_inst)
Out[4]: cfme.containers.template.Template

**Comment by shalomnaim1** [comment]
@mfalesni PR title changed as requested, waiting for review

File: cfme/tests/containers/test_smart_management.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+               "Template": lambda row: {"name": row.name.text, "provider": provider},
+               "Image": lambda row: {"name": row.name.text,
+                                     "tag": row.tag.text, "provider": provider},
+               "Image_Registry": lambda row: {"host": row.host.text, "provider": provider}}
+
+    return obj_creator(**factory[obj_creator.__module__.title().split(".")[-1]](row))
+
+
+@pytest.mark.parametrize('test_item',
+                         TEST_ITEMS, ids=[testItem.args[1].pretty_id() for testItem in TEST_ITEMS])
```
Comment (mfalesni) [inline cfme/tests/containers/test_smart_management.py]: Use snake_case instead of camelCase

File: cfme/tests/containers/test_smart_management.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    obj_inst.summary.reload()
+    tag_display_text = obj_inst.summary.smart_management.my_company_tags.pop()
+    tag_display_text = tag_display_text.text_value
+
+    assert re.match(regex, tag_display_text), "Tag formatting is invalid! "
+    actual_tags_on_instance = obj_inst.get_tags()
+
+    # Validate tag seted successfully
+    assert len(actual_tags_on_instance) == 1, "Fail to set a tag for {obj_type}".format(
+        obj_type=test_item.obj.__module__.title().split(".")[-1])
```
Comment (mfalesni) [inline cfme/tests/containers/test_smart_management.py]: I see multiple uses of this code, please DRY it out in a function or something.

File: cfme/tests/containers/test_smart_management.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    return obj_creator(**factory[obj_creator.__module__.title().split(".")[-1]](row))
+
+
+@pytest.mark.parametrize('test_item',
+                         TEST_ITEMS, ids=[testItem.args[1].pretty_id() for testItem in TEST_ITEMS])
+def test_smart_management_add_tag(provider, test_item):
+
+    # Select random instanse from instanse table
+    chosen_row = navigate_and_get_rows(provider, test_item.obj, 1).pop()
```
Comment (mfalesni) [inline cfme/tests/containers/test_smart_management.py]: The comment does not reflect reality, I assume that this pops the last line in the table, I would expect you would use random.choice instead of pop.

Comment (shalomnaim1) [inline cfme/tests/containers/test_smart_management.py]: navigate_and_get_rows is a function that navigateing to some object page and get few random rows as requested.

In this case i needed only one row, so the function returns list with only one value.
to reduce the amount of code and un-needed variables I decided to pop out the one value on the same line.

Comment (mfalesni) [inline cfme/tests/containers/test_smart_management.py]: Yes, sure, then the name of navigate_and_get_rows should actually be navigate_and_get_random_rows :) (though that is out of scope for this PR)

File: cfme/tests/containers/test_smart_management.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+from cfme.containers.image import Image
+from cfme.containers.project import Project
+from cfme.containers.node import Node
+from cfme.containers.image_registry import ImageRegistry
+from cfme.containers.pod import Pod
+from cfme.containers.template import Template
+from cfme.containers.provider import navigate_and_get_rows
+
+pytestmark = [
+    pytest.mark.uncollectif(lambda: current_version() < "5.6"),
```
Comment (izapolsk) [inline cfme/tests/containers/test_smart_management.py]: OPTIONAL: this is unneccessary now since 5.6 was excluded from testing scope.

**Comment by izapolsk** [comment]
PRT failed against 5.7. I've restarted PRT and will merge this PR when PRT are done.

**Comment by shalomnaim1** [comment]
@izapolsk all green on PRT, please merage
Thanks :-)

**Comment by izapolsk** [comment]
@shalomnaim1, nope, not green. PRT were skipped against all providers because there was some issue with provider setup. Please check what's happened.
I'll be happy to merge it once PRT are green.

**Comment by shalomnaim1** [comment]
@izapolsk Can you check now? i think everything passed.
sorry for nagging  :-)
